src/data/fintype/basic.lean

```diff
@@ -1500,6 +1510,11 @@ assume (hf : surjective f),
 have H : infinite α := infinite.of_surjective f hf,
 @infinite.not_fintype _ H infer_instance
 
+instance function.embedding.fintype' {α β} [infinite α] [fintype β] : fintype (α ↪ β) :=
```

Comment:
Can you please add

```
noncomputable def fintype_of_not_infinite {α : Type*} (h : ¬ infinite α) : fintype α :=
begin
  rw [←not_nonempty_fintype, not_not] at h,
  exact classical.choice h,
end

noncomputable
instance function.embedding.fintype'' {α β : Type*} [fintype β] : fintype (α ↪ β) :=
begin
  by_cases h : infinite α,
  { resetI, apply_instance, },
  { haveI := fintype_of_not_infinite h, classical, apply_instance, },
end
```

Reply:
The proof of `fintype.exists_infinite_fiber` should be refactored to use `fintype_of_not_infinite`.

Reply:
should I put a lower instance priority on the `noncomputable` instance or will it do so automatically?

Reply:
I've run into pain in the past with noncomputable fintype instances, and dont really think they are a good idea. For one, they prevent lean from evaluating sums over small embedding types

Reply:
is it worth possibly not making these an explicit instance, but an available defn for people to use if required?

Reply:
Hmm... I'm skeptical about avoiding classical `fintype` instances in principle. It's not very mathlib-y. :-)

Unless it actually breaks something already in mathlib, I'd propose punting on any computational problems. Perhaps, @eric-wieser, you could explain your explain over on zulip?